pmd-core/src/test/java/net/sourceforge/pmd/renderers/SummaryHTMLRendererTest.java

```diff
@@ -31,6 +31,11 @@ public Renderer getRenderer() {
         return result;
     }
 
+    @Override
+    protected String getSourceCodeFilename() {
+        return "notAvailable";
```

Comment:
Did you need to override this to get it working? Would it work with the super method when the source code filename is "notAavailable.ext"?

Reply:
It would not work with the super method. The final file name in the source code then would be *link_prefix**notAvilable.ext**.html#line_prefix1* which causes the Renderer to throw up since it then assumes `ext.html` is the file extension.

Reply:
```[ERROR] net.sourceforge.pmd.renderers.SummaryHTMLRendererTest.testRendererMultiple(net.sourceforge.pmd.renderers.SummaryHTMLRendererTest)
[ERROR]   Run 1: SummaryHTMLRendererTest>AbstractRendererTest.testRendererMultiple:125 expected:<..._prefixnotAvailable.[ext.html#line_prefix1">notAvailable.ext</a></td>
<td align="center" width="5%">1</td>
<td width="*">blah</td>
</tr>
<tr> 
<td align="center">2</td>
<td width="*%"><a href="link_prefixnotAvailable.ext.html#line_prefix1">notAvailable.ext]</a></td>
<td align...> but was:<..._prefixnotAvailable.[html#line_prefix1">notAvailable</a></td>
<td align="center" width="5%">1</td>
<td width="*">blah</td>
</tr>
<tr> 
<td align="center">2</td>
<td width="*%"><a href="link_prefixnotAvailable.html#line_prefix1">notAvailable]</a></td>
<td align...>
[ERROR]   Run 2: SummaryHTMLRendererTest>AbstractRendererTest.testRendererMultiple:125 expected:<..._prefixnotAvailable.[ext.html#line_prefix1">notAvailable.ext</a></td>
<td align="center" width="5%">1</td>
<td width="*">blah</td>
</tr>
<tr> 
<td align="center">2</td>
<td width="*%"><a href="link_prefixnotAvailable.ext.html#line_prefix1">notAvailable.ext]</a></td>
<td align...> but was:<..._prefixnotAvailable.[html#line_prefix1">notAvailable</a></td>
<td align="center" width="5%">1</td>
<td width="*">blah</td>
</tr>
<tr> 
<td align="center">2</td>
<td width="*%"><a href="link_prefixnotAvailable.html#line_prefix1">notAvailable]</a></td>
<td align...>
```
If the sourceFilename has an extension, it is apprently stripped away by the renderer